### PR TITLE
Allow pyside2 gui loop

### LINKED ISSUE #398: os.environ['QT_API'] gets overwritten in event-loop. No pyside2 integration.
Consider this IPython interactive session:

```python
Jupyter QtConsole 4.5.0.dev0
Python 3.6.8 |Anaconda, Inc.| (default, Feb 21 2019, 18:30:04) [MSC v.1916 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.3.0.dev -- An enhanced Interactive Python. Type '?' for help.

In [1]: IP = get_ipython()

In [2]: import os

In [3]: os.environ['QT_API']
Out[3]: 'pyside2'

In [4]: IP.enable_gui('qt')

In [5]: os.environ['QT_API']
Out[5]: 'pyqt5'

In [6]: os.environ['QT_API'] = 'pyside2'

In [7]: os.environ['QT_API']
Out[7]: 'pyqt5'

In [8]: from IPython.lib.guisupport import get_app_qt4

In [9]: app = get_app_qt4()

In [10]: app
Out[10]: <PySide2.QtWidgets.QApplication at 0x7ff9ddd4c568>
```

What happened between 6 and 7?

The answer is that os.environ['QT_API'] gets overwritten in the event-loop all the time once it has been activated. And even though I initially selected pyside2 it changes the env-var to pyqt5. But apparently the desired backend (pyside2) is still used. I do however run into problems, e.g. when importing directly from PyQt5 versus PySide2 depending on the environment variable.

I will attempt a fix.

### PR DESCRIPTION
**Pull request by bergtholdt**
Removed writing to the environment inside GUI loop.

Fixes #398 

Makes #399 obsolete.

### CODE REVIEW

File: ipykernel/eventloops.py [hunk 1]
```diff
@@ -125,7 +125,6 @@ def loop_qt4(kernel):
 @register_integration('qt', 'qt5')
 def loop_qt5(kernel):
     """Start a kernel with PyQt5 event loop integration."""
-    os.environ['QT_API'] = 'pyqt5'
```
Comment (ccordoba12) [inline ipykernel/eventloops.py]: If this is not set, then there's no way to select the right Python bindings to create the event loop (unless users set `QT_API` by hand).

Therefore, I'm -1 on this one.

Comment (bergtholdt) [reply ipykernel/eventloops.py]: OK, changed it to set if unset. This also prevents it from being overwritten in every loop (as it was originally).

**Comment by ccordoba12** [comment]
Ok, this is much better now, thanks! The only problem I see with this is that users still need to set `QT_API='pyside2'` by hand before running `%gui qt5`, even if PySide2 is the only Qt binding installed in your conda/virtual env.

@bergtholdt, do you think we could improve that situation?

**Comment by bergtholdt** [comment]
OK I changed it to use pyqt5 as default and pyside2 as fall-back. That should not affect anything that worked before.

A little improvement, in the case that both backends are available, could be done to check if either module has been imported before calling %gui qt5 by checking sys.modules, but I would suspect that in that case the user sets the variable herself ("explicit is better than implicit").

**Comment by ccordoba12** [comment]
> A little improvement, in the case that both backends are available, could be done to check if either module has been imported before calling %gui qt5 by checking sys.modules, but I would suspect that in that case the user sets the variable herself ("explicit is better than implicit").

Ok, let's go with this simple approach for now and see how users respond to it.

File: ipykernel/eventloops.py [hunk 2]
```diff
@@ -125,7 +125,16 @@ def loop_qt4(kernel):
...
+    if os.environ.get('QT_API', None) is None:
+        try:
+            import PyQt5
+            os.environ['QT_API'] = 'pyqt5'
+        except ImportError:
+            try:
+                import PySide2
+                os.environ['QT_API'] = 'pyside2'
+            except ImportError:
+                pass
```
Comment (ccordoba12) [inline ipykernel/eventloops.py]: I think we should set `QT_API='pyqt5'` at this point to have the same behavior as before?

Comment (ivanov) [reply ipykernel/eventloops.py:137]: I'm super necromancing this PR - but are there circumstances where there's any effect to setting `QT_API='pyqt5'` even though both   `import PyQt5` and `import pyside2` have failed? If so, then we can add that in place of the path, but I'm thinking maybe this is OK as is?

**Comment by ivanov** [comment]
Hey there, @bergtholdt, thank you for your contribution. I'm going through old PRs and this one has stalled.

I'm going to power cycle this one (close it and reopen it) just so we get the latest set of tests to rerun on it, and once I hear back on my the last remaining question to @ccordoba12 (or anyone else knowledgeable in QT stuff), then we can merge it.

**Comment by ivanov** [comment]
@ccordoba12 and @stukowski - does 2d15ed6a4aad182fd0ba276cd7058c90155bfba3 look like I'm doing the thing you were talking about to keep with previous behavior? I'll wait to merge until I hear back or someone more competent shows up :smile:

**Comment by ccordoba12** [comment]
> does 2d15ed6 look like I'm doing the thing you were talking about to keep with previous behavior?

Yeah, I think it does. I mean, it fallbacks to setting `QT_API` to `pyqt5` if we fail to import PySide2, which should preserve the previous behavior at the very least.

So I'm ok with merging this one now.

**Comment by ivanov** [comment]
Perfect, thank you, Carlos!

**Comment by ivanov** [comment]
and thank you @bergtholdt  for writing this up in the first place, apologies that it took so long to get merged in.

**Comment by ccordoba12** [comment]
> Perfect, thank you, Carlos!

No prob, and thanks to you for persevering until this PR was merged.

**Pull request merged**